### PR TITLE
E2E Upgrade Part 2 - Human Readable IBC Denom

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #4216

This PR adds a helper fn which asserts the human readable denom is present

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/transfer/base_test.go [hunk 1]
```diff
@@ -103,6 +103,10 @@ func (s *TransferTestSuite) TestMsgTransfer_Succeeds_Nonincentivized() {
 
 		expected := testvalues.IBCTransferAmount
 		s.Require().Equal(expected, actualBalance.Int64())
+
+		if testvalues.HumanReadableDenomFeatureReleases.IsSupported(chainBVersion) {
+			s.AssertHumanReadableDenom(ctx, chainB, chainADenom, channelA)
```
Comment (chatton) [inline e2e/tests/transfer/base_test.go:108]: cc @crodriguezvega

File: e2e/testvalues/values.go [hunk 2]
```diff
@@ -108,3 +108,8 @@ var LocalhostClientFeatureReleases = semverutil.FeatureReleases{
 		"v7.1",
 	},
 }
+
+// HumanReadableDenomFeatureReleases represents the releases the human readable denom feature was released in.
+var HumanReadableDenomFeatureReleases = semverutil.FeatureReleases{
```
Comment (damiannolan) [inline e2e/testvalues/values.go:113]: could also call this `DenomMetadataFeatureReleases` might be the tiniest bit less wordy, but whatever, happy with this too!

**Review by damiannolan — APPROVED** [review]
LGTM!

File: e2e/testvalues/values.go (hunk 2 above)
Comment (chatton) [reply e2e/testvalues/values.go:113]: I'll maybe do the rename in #4660 to avoid needing to do a full re-run of the upgrades here!

Comment (chatton) [reply e2e/testvalues/values.go:113]: nvm the upgrade test failed anyway :D

**Comment by chatton** [comment]
@damiannolan, I accidentally reverted a change I made previously. I updated the test to upgrade chain B (so we can test for the denom on chain B based on how we did the ibc transfer)

File: e2e/tests/transfer/base_test.go (hunk 1 above)
Comment (crodriguezvega) [reply e2e/tests/transfer/base_test.go:110]: I put together both assertions (check metadata is stored, check that balance returns the human readable denom). I hope that's ok, @chatton.

File: e2e/testsuite/grpc_query.go [hunk 3]
```diff
@@ -394,3 +383,30 @@ func (s *E2ETestSuite) QueryGranterGrants(ctx context.Context, chain *cosmos.Cos
 
 	return grants.Grants, nil
 }
+
+// QueryBalances returns all the balances on the given chain for the provided address.
+func (s *E2ETestSuite) QueryAllBalances(ctx context.Context, chain ibc.Chain, address string, resolveDenom bool) (sdk.Coins, error) {
```
Comment (crodriguezvega) [inline e2e/testsuite/grpc_query.go:388]: I moved this down here to keep both usages of the bank query client together.

File: .github/workflows/e2e-upgrade.yaml [hunk 4]
```diff
@@ -71,7 +71,7 @@ jobs:
       chain-binary: simd
       chain-a-tag: v7.0.0
       chain-b-tag: v7.0.0
-      chain-upgrade-tag: pr-4591 # TODO: update this to a real tag once v8 is released
+      chain-upgrade-tag: main # TODO: update this to a real tag once v8 is released
```
Comment (crodriguezvega) [inline .github/workflows/e2e-upgrade.yaml:74]: I opened https://github.com/cosmos/ibc-go/issues/4674.

**Review by crodriguezvega — APPROVED** [review]
Thank you, @chatton!

File: e2e/testvalues/values.go [hunk 5]
```diff
@@ -48,11 +48,6 @@ func SolomachineClientID(id int) string {
 	return fmt.Sprintf("06-solomachine-%d", id)
 }
 
-// TokenMetadataFeatureReleases represents the releases the token metadata was released in.
-var TokenMetadataFeatureReleases = semverutil.FeatureReleases{
```
[the remark this answers is not in the document]
Comment (chatton) [reply e2e/testvalues/values.go]: whoops I missed this and deleted the one I added and use the one you used 😄

**Pull request merged**